src/main/java/com/github/devotedmc/hiddenore/Config.java

```diff
@@ -47,24 +51,51 @@ private Config() {
 		ignoreSilktouch = false;
 		defaultPrefix = "You found hidden ore!";
 		transformAttemptMultiplier = 3;
+		configurations = new HashMap<String, FileConfiguration>();
 	}
 
 	public static void loadConfig() {
 		try {
-			file = HiddenOre.getPlugin().getConfig();
-			doLoad();
+			mainConfig = HiddenOre.getPlugin().getConfig();
+			doLoadMainConfig(mainConfig);
+			for (World world : HiddenOre.getPlugin().getServer().getWorlds()) {
+				if(world == null) {
+					WorldCreator wc = new WorldCreator(world.getName());
+					HiddenOre.getPlugin().getServer().createWorld(wc);
+				}
+				Reader reader = new InputStreamReader(HiddenOre.getPlugin().getResource("default-world.yml"));
+				File saveFile = new File(HiddenOre.getPlugin().getDataFolder(), String.format("%s-config.yml", world.getName()));
+				File defaultWorldFile = new File(HiddenOre.getPlugin().getDataFolder(), "default-world.yml");
+				configurations.put(world.getName(), YamlConfiguration.loadConfiguration(reader));
+				if (saveFile.exists()) {
+					configurations.get(world.getName()).load(saveFile);
+				} else {
+					HiddenOre.getPlugin().saveResource("default-world.yml", false);
+					if (defaultWorldFile.exists()) {
+						defaultWorldFile.renameTo(saveFile);
+						configurations.get(world.getName()).set("world_name", world.getName());
+						configurations.get(world.getName()).save(saveFile);
+					}
+					else {
+						HiddenOre.getPlugin().getLogger().warning(String.format("A YAML file could not be created for the world named %s.", world.getName()));
+					}
+				}
+				doLoadWorldConfig(configurations.get(world.getName()));
+			}
+
 		} catch (Exception e) {
-			HiddenOre.getPlugin().getLogger().log(Level.WARNING, "An error occured while loading config!", e);
+			HiddenOre.getPlugin().getLogger().log(Level.WARNING, "An error occurred while loading config!", e);
 		}
 	}
 
-	public static void doLoad() {
+	public static void doLoadMainConfig(FileConfiguration file) {
 		Config i = new Config();
 
 		isDebug = file.getBoolean("debug", isDebug);
 		caveOres = file.getBoolean("caveOres", caveOres);
-
-		trackFileName = file.getString("track_file", trackFileName);
+		if (file.isSet("track_file")) {
```

Comment:
This is superfluous when using `getString(key, default)` as if it doesn't exist, default is used instead, no need to "pre-check".